common/browser_process.go

```diff
@@ -186,37 +181,48 @@ func execute(
 		}
 	}()
 
-	return cmd, stdout, done, nil
+	return cmd, done, nil
 }
 
-// parseDevToolsURL grabs the websocket address from chrome's output and returns it.
-func parseDevToolsURL(ctx context.Context, rc io.Reader) (wsURL string, _ error) {
-	type result struct {
-		devToolsURL string
-		err         error
-	}
-	c := make(chan result, 1)
-	go func() {
-		const prefix = "DevTools listening on "
-
-		scanner := bufio.NewScanner(rc)
-		for scanner.Scan() {
-			if s := scanner.Text(); strings.HasPrefix(s, prefix) {
-				c <- result{
-					strings.TrimPrefix(strings.TrimSpace(s), prefix),
-					nil,
-				}
-				return
-			}
+// getDevToolsURL returns the DevTools WebSocket address by reading the
+// DevToolsActivePort file in the data directory.
+func getDevToolsURL(dataDir string) (wsURL string, rerr error) {
+	var (
+		f                  *os.File
+		fpath              = filepath.Join(dataDir, "DevToolsActivePort")
+		maxReadAttempts    = 10
+		readAttemptDelayMS = 50
```

Comment:
Yeah, `test-tip` failed with `panic: launching browser: getting DevTools URL: unable to read file "/tmp/xk6-browser-data-2308635430/DevToolsActivePort" in 500ms`. :disappointed:

I guess we could just increase it to 5s total to be safe (so 100 attempts)? Even 1 or 2s might be too low for some machines.